src/linux.rs

```diff
@@ -0,0 +1,143 @@
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::{io::Error, net::IpAddr};
+
+use neli::{
+    attr::Attribute,
+    consts::{
+        nl::{NlmF, NlmFFlags},
+        rtnl::{
+            Arphrd, IffFlags, Ifla, RtAddrFamily, RtScope, RtTable, Rta, Rtm, RtmFFlags, Rtn,
+            Rtprot,
+        },
+        socket::NlFamily,
+    },
+    nl::{NlPayload, Nlmsghdr},
+    rtnl::{Ifinfomsg, Rtattr, Rtmsg},
+    socket::NlSocketHandle,
+    types::RtBuffer,
+};
+
+use crate::default_err;
+
+const fn family(remote: &IpAddr) -> RtAddrFamily {
+    if remote.is_ipv4() {
+        RtAddrFamily::Inet
+    } else {
+        RtAddrFamily::Inet6
+    }
+}
+
+const fn addr_len(remote: &IpAddr) -> u8 {
+    if remote.is_ipv4() {
+        32
+    } else {
+        128
+    }
+}
+
+fn addr_bytes(remote: &IpAddr) -> Vec<u8> {
+    match remote {
+        IpAddr::V4(ip) => ip.octets().to_vec(),
+        IpAddr::V6(ip) => ip.octets().to_vec(),
+    }
+}
+
+pub fn interface_and_mtu_impl(remote: IpAddr) -> Result<(String, usize), Error> {
+    // Create a netlink socket.
+    let mut socket = NlSocketHandle::connect(NlFamily::Route, None, &[])?;
```

Comment:
Do I understand correctly that a connect on a netlink socket would still be blocklisted?

https://github.com/mozilla/neqo/pull/2135/files/5f03837fb4d97906be80109b6b8501f2f31b3aa8#r1781206784

To avoid confusion, if so, I don't think we should work around the blocklist, but change it instead.

Reply:
New code only uses `socket`, so that is hopefully OK.